MicrosoftDefender/microsoftdefender_modules/action_push_indicators.py

```diff
@@ -0,0 +1,138 @@
+from collections import defaultdict
+from typing import Any
+from urllib.parse import urljoin
+
+from requests import Response
+
+from .action_base import MicrosoftDefenderBaseAction
+from .helpers import stix_to_indicators
+
+
+class PushIndicatorsAction(MicrosoftDefenderBaseAction):
+    DEFAULT_SEKOIA_BASE_URL = "https://app.sekoia.io"
+    SUPPORTED_TYPES_MAP: dict[str, dict[str, str]] = {
+        "ipv4-addr": {"value": "IpAddress"},
+        "ipv6-addr": {"value": "IpAddress"},
+        "domain-name": {"value": "DomainName"},
+        "file": {"hashes.MD5": "FileMd5", "hashes.SHA1": "FileSha1", "hashes.sha256": "FileSha256"},
+        "url": {"value": "Url"},
+    }
+    """
+    SUPPORTED_TYPES_MAP is used to define the mapping from STIX pattern
+    to the different IOCs types supported by Crowdstrike Falcon.
+    "stix_root_key": {"stix_sub_key": "target_type"}
+    Example with ipv4:
+    Mapping with "ipv4-addr": {"value": "ipv4"} for IOC "[ipv4-addr:value = 'X.X.X.X']"
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.sekoia_base_url = self.DEFAULT_SEKOIA_BASE_URL
+
+    def list_indicators(self, q: str | None = None) -> Response:
+        url = urljoin(self.client.base_url, "api/indicators")
+        if q:
+            url = url + "?$filter=" + q
+
+        return self.client.get(url)
+
+    def import_indicators(self, indicators: list[dict[str, Any]]) -> Response:
+        url = urljoin(self.client.base_url, "api/indicators/import")
+        return self.client.post(url, json={"Indicators": indicators})
+
+    def delete_indicators_by_ids(self, indicators_ids: list[str]) -> Response:
+        url = urljoin(self.client.base_url, "api/indicators/BatchDelete")
+        return self.client.post(url, json={"IndicatorIds": indicators_ids})
+
+    def remove_indicators(self, indicators: list[dict[str, Any]]) -> None:
+        ids_to_remove: list[Any] = []
+
+        for indicator in indicators:
+            q = f"indicatorValue+eq+'{indicator['indicatorValue']}'"
+            response = self.list_indicators(q=q)
+            self.process_response(response)
```

Comment:
I wonder if we can't use the [`in`](https://docs.oasis-open.org/odata/new-in-odata/v4.01/cn04/new-in-odata-v4.01-cn04.html#_Toc21700044) operator?